src/promptflow-core/promptflow/executor/flow_executor.py

```diff
@@ -847,7 +847,7 @@ def _exec_inner_with_trace(
     ):
         # need to get everytime to ensure tracer is latest
         otel_tracer = otel_trace.get_tracer("promptflow")
-        with otel_tracer.start_as_current_span(self._flow.name) as span:
+        with otel_tracer.start_as_current_span(self._flow.name) as span, self._record_keyboard_interrupt_to_span(span):
```

Comment:
Do we do this only for "Flow" spans? How about a flex flow that does not have a "Flow" span?

Reply:
Good catch. Maybe open another PR for flex scenario.
This PR is enough dangerous.

Reply:
Another PR: https://github.com/microsoft/promptflow/pull/2918